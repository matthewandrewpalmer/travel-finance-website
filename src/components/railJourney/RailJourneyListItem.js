import React from "react";
import Card from "@material-ui/core/Card";
import CardActionArea from "@material-ui/core/CardActionArea";
import CardContent from "@material-ui/core/CardContent";
import CardActions from "@material-ui/core/CardActions";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";
import makeStyles from "@material-ui/core/styles/makeStyles";
import DirectionsBusOutlinedIcon from "@material-ui/icons/DirectionsBusOutlined";
import ExploreIcon from "@material-ui/icons/Explore";
import dayjs from "dayjs";
import CardMedia from "@material-ui/core/CardMedia";

const useStyles = makeStyles({
    card: {
        // maxWidth: 345,
        margin: 25,
        // width: "100%"
    },
});


function RailJourneyListItem(props) {
    const styles = useStyles();

    let index = props.index;
    let journey = props.journey;


    function journeyTitle() {
        switch (journey.journey_type) {
            case "PlusBus":
                return (
                    <Typography gutterBottom variant="h5" component="h2">
                        <DirectionsBusOutlinedIcon/> PlusBus {journey.departing}
                    </Typography>
                );
            case "Rover":
                return (
                    <Typography gutterBottom variant="h5" component="h2">
                        <ExploreIcon/> {journey.ticket_name.String}
                    </Typography>
                );
            default:
                return (
                    <Typography gutterBottom variant="h5" component="h2">
                        {journey.departing} - {journey.destination.String}
                    </Typography>
                );
        }
    }


    return (
        <Card key={index} className={styles.card}>
            <CardActionArea>
                <CardMedia
                    component="img"
                    alt="Contemplative Reptile"
                    height="50"
                    image="/class_800.png"
                    title="Contemplative Reptile"
                />
                <CardContent>
                    {journeyTitle()}
                    <Typography variant="body2" color="textSecondary" component="p">
                        Ticket Type: {journey.journey_type}
                        <br/>
                        Cost: Â£{(Math.round(journey.cost * 100) / 100).toFixed(2)}
                        <br/>
                        Date: {dayjs(journey.date).format("DD/MM/YYYY")}
                    </Typography>
                </CardContent>
            </CardActionArea>
            <CardActions>
                <Button size="small" color="primary">
                    Share
                </Button>
                <Button size="small" color="primary">
                    Learn More
                </Button>
            </CardActions>
        </Card>
    );
}

export default RailJourneyListItem;