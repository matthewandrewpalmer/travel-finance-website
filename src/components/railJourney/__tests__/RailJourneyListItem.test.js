import React from "react";
import RailJourneyListItem from "../RailJourneyListItem";
import renderer from "react-test-renderer";
import {render} from "@testing-library/react";

describe("RailJourneyListItem", () => {

    const testProps = {
        journey: {
            id: 0,
            journey_type: "Single",
            departing: "departing",
            destination: {String: "destination ", Valid: true},
            ticket_name: {String: "", Valid: false},
            date: "2020-04-15T00:00:00Z",
            railcard_used: true,
            cost: 100,
            total_cost: 10,
        },
        index: 0
    };

    it("renders correctly", () => {
        const tree = renderer.create(
            <RailJourneyListItem
                {...testProps}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    test.skip("renders departing and destination", () => {
        const {getByText} = render(
            <RailJourneyListItem
                {...testProps}
            />
        );
        let string = testProps.journey.departing + "\n - \n" + testProps.journey.destination;
        const linkElement = getByText(string);
        expect(linkElement).toBeInTheDocument();
    });
});