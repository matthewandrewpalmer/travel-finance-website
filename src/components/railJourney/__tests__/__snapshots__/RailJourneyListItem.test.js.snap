// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RailJourneyListItem renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        departing
         - 
        destination 
      </h2>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Ticket Type: 
        Single
        <br />
        Cost: Â£
        100.00
        <br />
        Date: 
        15/04/2020
      </p>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight"
    />
  </button>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Share
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Learn More
      </span>
    </button>
  </div>
</div>
`;
